File: src/4-Problem01.py
#Prisila Michelle - 13516129

n = int(input("Masukkan jumlah barang : "))

barang = []
for i in range(0,n):
	x = input("Masukkan nama barang : ")
	barang.append(x)

while barang:
	print("	1. Jual")
	print("	2. Lihat stok")

	menu = input("Pilihan menu : ")

	if menu == '1':
		chosenthing = input("Masukkan nama barang : ")
		found = False
		i = 0
		while i < len(barang) and not found:
			if barang[i] == chosenthing:
				found = True
			else:
				i = i + 1

		if found:
			print("Barang berhasil dijual")
			barang.remove(chosenthing)
		else:
			print("Barang tidak ada")

	elif menu == '2':
		print("Stok toko : ", end='')
		for i in range(len(barang)):
			if i < (len(barang)-1):
				print(barang[i], end=', ')
			else:
				print(barang[i])

print("Barang sudah habis, silahkan melakukan restok barang")

File: src/4-Problem01.pas
//Nama / NIM : Christian Kevin Saputra / 13516073

Program Bab4_Array;

var
	stok , temp: array of string;
	jumlah,i,j,k,pilihan :integer;
	found: boolean;
	namabarang :string;

{Algoritma}
Begin
	write('Masukkan jumlah barang: ');
	readln(jumlah);
	SetLength(stok, jumlah);
	for i:=0 to jumlah-1 do
	Begin
		write('Masukkan nama barang: ');
		readln(stok[i]);
	end;
	while length(stok) <> 0 do
	Begin
		writeln();
		writeln('1.Jual');
		writeln('2.Lihat stok');
		writeln();
		write('Pilihan menu: ');
		readln(pilihan);
		while (pilihan < 1) or (pilihan > 2) do
		Begin
			writeln('Pilihan menu salah. Silahkan ulangi');
			write('Pilihan menu: ');
			readln(pilihan);
		end;
		j:=0;
		if (pilihan = 1) then
		Begin
			found:=false;
			write('Masukkan nama barang:');
			readln(namabarang);
			while (j <= length(stok) - 1) and (not(found)) do
			Begin
				if (stok[j] = namabarang) then
				Begin
					stok[j] := 'null';
					found:=true;
					//mengurangi elemen di dalam array
					SetLength(temp, length(stok)-1);
					k:=0;
					for i:=0 to length(stok) - 1 do
					Begin
						if (i <> j) then
						Begin
							temp[k] := stok[i];
							k:= k + 1;
						end;
					end;		
					stok:=Copy(temp,0,length(temp));		
				end else
					j := j + 1;
			end;

			if (not(found)) then
				writeln('Barang tidak ada')
			else
				writeln('Barang berhasil dijual');
		end else
		Begin
			write ('Stok toko: ');
			while (j < length(stok)-1) do
			Begin
				if(stok[j] = 'null') then
					j:= j + 1
				else
					write(stok[j],', ');
					j:= j+1;
			end;
			if(stok[j] <> 'null') then
			Begin
				writeln(stok[j]);
				j:= j+1;
			end;
		end;
	end;
	writeln('Barang sudah habis, silahkan melakukan restok barang');
end.